mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvpManager.java

```diff
@@ -56,28 +56,51 @@
 	 * @param id represents the kit ID
 	 * @return the kit from cache or table
 	 */
-	Kit getKit(Transaction transaction, int id) {
-		return existingKits.get(id, num -> {
-			var jooq = transaction.getProperty(DSLContext.class);
+	Kit getKitById(Transaction transaction, int id) {
+		var jooq = transaction.getProperty(DSLContext.class);
```

Comment:
The changes you made to this method weaken the effectiveness of the caching.

If the kit exists in the cache, your implementation should not query the database. Check the in-memory cache before you perform any queries. In other words, you should have `existingKits.get` as the first call in the method.

Reply:
ah, wait i completely destroyed the use of Cache 😂 .... I'll change that sorry